lib/autobuild/import/archive.rb

```diff
@@ -259,10 +259,33 @@ def download_from_url(package)
         # @return [Boolean] true if a new file was downloaded, false otherwise
         def update_cache(package)
             updated = download_from_url(package)
-            @cachefile_digest = Digest::SHA1.hexdigest File.read(cachefile)
+            @cachefile_digest = read_cachefile_digest
+
+            if @expected_digest and @expected_digest != @cachefile_digest
```

Comment:
Please use `&&` instead of `and`, this is a lot more idiomatic in Ruby

Reply:
I've changed it